src/chapters/thymeleaf-views/why-templates.rst

```diff
@@ -0,0 +1,145 @@
+The content includes text, images, a navigation bar, a search box, linked menu
+options at the bottom of the page, and 118 carefully colored boxes with
+links---one for each element on the periodic table. All of this content is
+very deliberately arranged and styled.
+
+Imagine if you were tasked by your boss to build this website. Setting up the
+HTML tags for the navigation bar would be straightforward, but what about the
+element boxes? You would need to make 118 similar structures, but with
+different text, links, and colors. Trying to make the table structure work
+would be tedious at best, and excruciatingly difficult at worst.
+
+Also, what if a new element gets discovered, or some of the data for the
+elements changes? Updating the text, colors, layout, etc. means adjusting those
+items in the HTML. Also, if that information appears in other areas of the
+website, then you need to modify that code as well.
+
+Whew! Changing the website rapidly becomes problematic, especially since it
+contains lots of data and consists of multiple pages. If only there was
+a way to automate the tasks required to build and maintain a website...
+
+Templates are Frameworks
+-------------------------
+
+.. index:: ! template
+
+A **template** provides the general structure for a web page. It identifies
+where different elements get placed on the page, but it does NOT fill them with
+content. Think of a template as an outline for what we want the page to look
+like. No details yet, just defined spaces where information needs to be added.
+
+Let's see how using a template makes our lives easier.
+
+No Template
+^^^^^^^^^^^^
+
+The code below displays a simple list. It defines the location for the heading
+and each ``<li>`` element, in addition to a couple of fun links. The CSS file
+(not shown) specifies the font, text size, colors, etc.
+
+.. sourcecode:: html
+   :linenos:
+
+   <body>
+      <h1>Java Types</h1>
+      <div class="coffeeList">
+         <ol>
+            <li>French Roast</li>
+            <li>Espresso</li>
+            <li>Kopi Luwak</li>
+            <li>Instant</li>
+         </ol>
+      </div>
+      <hr>
+      <div class="links">
+         <h2>Links</h2>
+         <a href="https://www.launchcode.org/">LaunchCode</a> <br>
+         <a href="https://en.wikipedia.org/wiki/Coffee">Coffee</a>
+      </div>
+   </body>
+
+We could drastically improve the appearance and content of the page by playing
+around with the tags, classes, styles and text. However, any change we want to
+make needs to be coded directly into the HTML and CSS files, and this quickly
+becomes inefficient.
+
+A Better Way
+^^^^^^^^^^^^^
+
+Recall that a template represents a *view* in the MVC world. It sets up a
+structure to display the data delivered by the controller. The template itself
+contains *blanks* which get filled in by the data, and this information can
+change based on a user's actions.
+
+.. sourcecode:: guess
```

Comment:
I'm trying to avoid ``guess`` as much as possible, since I'm hesitant to rely on a black box.  ``html`` usually works for the Thymeleaf examples, and ``groovy`` is OK at highlighting Gradle code.

I had to use ``guess`` here because of the made up ``{templateInstructions}``.  The other code choices either threw errors or applied weird highlighting.